### ISSUE TITLE
The `findAll` API have something wrong!

### ISSUE THREAD

**Issue by Desgard** [cezheng/PySwiftyRegex#10]
```swift
// Regex
static let trendingDevListRegex = "<li class=\"d-sm-flex flex-justify(.|\n)+?</li>"
static let trendingDevRegex = "<?li class=\"d-sm-flex flex-justify(.|\n)+?id=\"pa-(.+?)\"(.|\n)+?</li>"
```

I use the `trendingDevListRegex` to pattern the [Github Trending - Developers](https://github.com/trending/developers) Page, and use the `trendingDevRegex` to pattern each one. And every match can be able to lose the first character just like the following screen shot:

![488fab0e-8c68-4e42-8625-73487299e560](https://user-images.githubusercontent.com/7804535/35262109-2bacf5b8-004e-11e8-8b8a-5c130af109f3.png)

It was very strange !! 🤣  (The bad network refuse me to upload the image..wait for some time.)

**Comment by cezheng** [comment]
Thanks for reporting the issue. Can you provide a snippet of runnable code so that I can easily reproduce it from Playground? (define a string literal that produces incorrect result, then call findall on it)

BTW, using regex on HTML is [never a good practice](https://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags) as it is very fragile and unreliable. You should use a XML/HTML parser and it will be a lot easier. You can take a look on my other project https://github.com/cezheng/Fuzi

**Comment by Desgard** [comment]
There are some 3rd framework in my source code, so I paste the code in this comment directly: 

```Swift
//
//  SepiTrendingHelper.swift
//  Sepicat
//
//  Created by Harry Twan on 23/01/2018.
//  Copyright © 2018 Desgard_Duan. All rights reserved.
//

import UIKit
import Alamofire
import PySwiftyRegex
import CocoaLumberjack

class SepiTrending: NSObject {
    static let shared = SepiTrending()
    
    struct SimpleDevModel {
        var login = ""
        var name = ""
        var avatar_url = ""
        var repo = ""
    }
    
    enum category {
        case repo
        case dev
    }
    
    struct const {
        static let trendingRepoUrl = "https://github.com/trending"
        static let trendingDevUrl  = "https://github.com/trending/developers"
        
        // Regex
        static let trendingDevListRegex = "<li class=\"d-sm-flex flex-justify(.|\n)+?</li>"
        static let trendingDevRegex = ".*?\"d-sm-flex flex-justify(.|\n)+?id=\"pa-(.+?)\"(.|\n)+?src=\"(https://avatars.+?)\".+?<span class=\"repo\".+?>(.+?)</span>.+</li>"
    }
    
    /// 获取 Trending User 列表
    public func fetchDeveloperTrendingList(completion: @escaping (Bool, [SimpleDevModel]?) -> Void) {
        guard let url = URL(string: const.trendingDevUrl) else {
            DDLogDebug("URL 解析错误")
            completion(false, nil)
            return
        }
        
        Alamofire.request(url, method: .get, parameters: nil, encoding: URLEncoding.default, headers: GithubManager.header)
            .responseData { response in
                switch response.result {
                case .success(let data):
                    guard let html = String(data: data, encoding: .utf8) else {
                        DDLogDebug("Data resolve error")
                        return
                    }
                    let res = self.resolve(html: html, type: .dev)
                    completion(true, res)
                    
                case .failure(let error):
                    DDLogDebug("Fetch trending error: \(error.localizedDescription)")
                    completion(false, nil)
                }
        }
    }
    
    /// Resolve String
    private func resolve(html: String, type: category) -> [SimpleDevModel] {
        let regex = re.compile(const.trendingDevListRegex)
        let subRegex = re.compile(const.trendingDevRegex)
        let ss = regex.findall(html).map { $0.replacingOccurrences(of: "\n", with: "") }
        var res: [SimpleDevModel] = []
        for s in ss {
            if let m = subRegex.match(s) {
                var thisOne = SimpleDevModel()
                thisOne.login = m.group(2) ?? ""
                thisOne.avatar_url = m.group(4) ?? ""
                if let repo = m.group(5) {
                    thisOne.repo = repo.replacingOccurrences(of: " ", with: "")
                }
                res.append(thisOne)
            }
        }
        return res
    }
}
```

**Comment by cezheng** [comment]
Turns out to be the same problem as #5, and Swift 4 migration tool incorrectly dropped the utf16. It should be fixed with 2.0.1. Please confirm.

**Comment by Desgard** [comment]
@cezheng Yeah! Thanks for the help 😊
